packages/components/checkbox/index.js

```diff
@@ -71,47 +77,49 @@ export const Checkbox = ({
       ]),
     },
     label: {
-      ml: isReversed ? '0' : '2',
-      mr: isReversed ? '2' : '0',
-      fontFamily: 'body',
-      fontSize: 'checkbox',
-      fontWeight: 'medium',
-      letterSpacing: 'checkbox',
-      color: `${
-        isDisabled ? 'font.checkbox.disabled' : 'font.checkbox.default'
-      }`,
+      ml: isReversed ? '0' : label.normal.margin,
+      mr: isReversed ? label.normal.margin : '0',
+      fontFamily: label.normal.fontFamily,
+      fontSize: label.normal.fontSize,
+      fontWeight: label.normal.fontWeight,
+      letterSpacing: label.normal.letterSpacing,
+      color: `${isDisabled ? label.disabled.color : label.normal.color}`,
       ...getAllowedProps(override.label, [...color, ...space, ...typography]),
     },
   };
 
   return (
-    <Box as="label" {...Override.wrapper}>
-      {/* This is the sibling input, it's visually hidden */}
-      <VisuallyHidden
-        as="input"
-        type="checkbox"
-        id={id}
-        name={name}
-        value={value}
-        checked={isChecked}
-        disabled={isDisabled}
-        onBlur={onBlur}
-        onFocus={onFocus}
-        onChange={onChange}
-        defaultChecked={defaultIsChecked}
-        {...Override.visuallyHidden}
-      />
+    <ThemeProvider theme={theme}>
+      <Box p="2">
+        <Box as="label" {...Override.wrapper}>
+          {/* This is the sibling input, it's visually hidden */}
+          <VisuallyHidden
+            as="input"
+            type="checkbox"
+            id={id}
+            name={name}
+            value={value}
+            checked={isChecked}
+            disabled={isDisabled}
+            onBlur={onBlur}
+            onFocus={onFocus}
+            onChange={onChange}
+            defaultChecked={defaultIsChecked}
+            {...Override.visuallyHidden}
+          />
 
-      {/* This is the control box with a check icon as children */}
-      <ControlBox {...Override.controlBox}>
-        <Icon name="customCheck" size="10px" />
```

Comment:
Shouldn't icon be custom ? I remember we had this in some commit. This is needed if we wish to change the check icon through props. Might change from project to project. May be even screens to screen. Let that prop be a renderer that can render SVGs

Reply:
If we are creating an Icon component there, we can allow the SVG a prop.
But here we are not going to allow the icon to be changed via prop because this is already as per the design.
This is the base design for the checkbox which should be as it is for a project IMO

Reply:
So for a new project we shouldn't be editing the source from `customCheck` to something right ? Also we cannot expect the user to save the new icon with title assigned to `customCheck`. They should name it in any name and pass it down here. But by default we can have `customCheck` as the icon name.

Reply:
If they wish to change the check icon, they can change the `path` in theme.js.

> So for a new project, we shouldn't be editing the source from `customCheck` to something right? 

No, we wont be editing here, only in the theme.js

> Also we cannot expect the user to save the new icon with title assigned to `customCheck`. They should name it in any name and pass it down here. But by default, we can have `customCheck` as the icon name.

We can write a doc on this, to change the `path` on `customCheck` from `theme.js`